Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,219 @@
+
+The layout is composed from a list of package names and versions specified as xml: 
+
+**Roslyn Example**
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+ <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis" Version="1.3.2" />
+    <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic" Version="1.3.2" />
+    <PackageReference Include=""Microsoft.CodeAnalysis.VisualBasic.Features" Version="1.3.2" />
+  </ItemGroup>
+</Project>
+```
+
+
+and issue a command like below:
+
+```
+dotnet cache --entries packages.xml --framework netcoreapp2.0 [--output C:\Foo] --runtime win7-x64 --framework-version 2.0.0-preview2-00001 [--no-optimize]
+
+--framework          Specifies the TFM that the package store is applicable to
+--output       The output directory to create the package store in (default: %USERPROFILE%\.dotnet or ~/.dotnet)
+
+--skip-optimization  Do not perform crossgen of the assemblies after "restore" (optimize is the default)
+
+--runtime          The runtime identifier of the target platform where these assemblies will be run
+--framework-version   The Microsoft.NETCore.App package version that will be used to run the assemblies
+
+```
+NOTE: It is a requirement that `packages.xml` is of msbuild format, because it forms the entry point from which the rest of the SDK's functionality can be accessed
+
+Hosting providers would create a `packages.xml` file corresponding to the packages that will be shared in their hosting environment and specify the file to `dotnet cache`. The file can be on the file system or from an URL. The TFM argument is used in the shared package layout described above.
+
+If `--optimize` is specified, we would precompile all the managed assets to native code in a temp folder before copying to the output folder. If crossgen is used, it would be the one acquired in the closure of the `Microsoft.NETCore.App` specified by the `--framework-version` option. Also, if no `--output` folder is specified, then the default is `~/.dotnet` or `%USERPROFILE%\.dotnet\`. The output asset files will be present in the following layout: `$HOME/.dotnet/packages/{tfm}/{package-name}/{package-version}/{asset-path}`.
+
+The output folder will be consumed by the runtime by adding to the `DOTNET_SHARED_PACKAGES_ROOTS` environment variable. See probe precedence below.
+
+# Building apps with shared packages
+
+The current mechanism to build applications that share assemblies is by not specifying a RID in the project file. Then, a portable app model is assumed and assemblies that are part of Microsoft.NETCore.App are found under the `dotnet` install root. With shared package store, applications have the ability to filter any set of packages from their publish output. Thus the decision of a portable or a standalone application is not made at the time of project authoring but is instead done at publish time. 
```

Comment:
Oh, I think the next paragraph takes care of that... It says this should be an error.